src/rust/src/x509/common.rs

```diff
@@ -391,27 +390,30 @@ pub(crate) fn parse_and_cache_extensions<
 >(
     py: pyo3::Python<'p>,
     cached_extensions: &mut Option<pyo3::PyObject>,
-    raw_exts: &Option<Extensions<'_>>,
+    raw_extensions: &Option<RawExtensions<'_>>,
```

Comment:
I think we actually want this method to take an `Option<Extensions>` and then the callers can obtain the extensions, via an `extensions()` method on the various `cryptography-x509` types. That way those types have the right methods that can be used from the validator crate.

Reply:
That SGTM, although if I understand correctly doing that will require us to make the `Iterator` change mentioned in  https://github.com/pyca/cryptography/pull/8892#discussion_r1189196107: without that change, we'd need to perform the fallible conversion from `RawExtensions -> Extensions` before `parse_and_cache_extensions`, defeating the point of having it inside of that function. Or did I miss something?

Reply:
Ok, let's leave this change for a seperate PR.